client/extensions/woocommerce/components/product-search/item.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import { fetchProductVariations } from 'woocommerce/state/sites/product-variations/actions';
+import formattedVariationName from 'woocommerce/lib/formatted-variation-name';
+import getKeyboardHandler from 'woocommerce/lib/get-keyboard-handler';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { getVariationsForProduct } from 'woocommerce/state/sites/product-variations/selectors';
+
+class ProductItem extends Component {
+	componentDidMount() {
+		const { siteId, productId } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchProductVariations( siteId, productId );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const { siteId, productId } = this.props;
+		if ( newProps.productId !== productId || newProps.siteId !== siteId ) {
+			this.props.fetchProductVariations( newProps.siteId, newProps.productId );
+		}
+	}
+
+	onClick = () => {
```

Comment:
We [should be avoiding](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#react-components) using names like `onClick` for handlers, perhaps `handleClick` may be better suited?